ios/FluentUI.Demo/FluentUI.Demo/Demos/DrawerVnextDemoController.swift

```diff
@@ -0,0 +1,174 @@
+
+        container.addArrangedSubview(UIView())
+
+        // Screen edge gestures to interactively present side drawers
+
+        let leadingEdgeGesture = UIScreenEdgePanGestureRecognizer(target: self, action: #selector(handleScreenEdgePan))
+        leadingEdgeGesture.edges = view.effectiveUserInterfaceLayoutDirection == .leftToRight ? .left : .right
+        view.addGestureRecognizer(leadingEdgeGesture)
+        navigationController?.navigationController?.interactivePopGestureRecognizer?.require(toFail: leadingEdgeGesture)
+
+        let trailingEdgeGesture = UIScreenEdgePanGestureRecognizer(target: self, action: #selector(handleScreenEdgePan))
+        trailingEdgeGesture.edges = view.effectiveUserInterfaceLayoutDirection == .leftToRight ? .right : .left
+        view.addGestureRecognizer(trailingEdgeGesture)
+
+        drawerController = DrawerVnext(contentViewController: DrawerContentController())
+        drawerController?.delegate = self
+    }
+
+    @objc private func showLeftDrawerClearBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = false
+            drawerController.drawerState.presentationDirection = .left
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func showLeftDrawerDimmedBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = true
+            drawerController.drawerState.presentationDirection = .left
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func showRightDrawerClearBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = false
+            drawerController.drawerState.presentationDirection = .right
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func showRightDrawerDimmedBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = true
+            drawerController.drawerState.presentationDirection = .right
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func handleScreenEdgePan(gesture: UIScreenEdgePanGestureRecognizer) {
+        guard gesture.state == .began else {
+            return
+        }
+
+        let leadingEdge: UIRectEdge = view.effectiveUserInterfaceLayoutDirection == .leftToRight ? .left : .right
+        if leadingEdge == .left {
+            self.showRightDrawerDimmedBackgroundButtonTapped()
+        } else {
+            self.showLeftDrawerClearBackgroundButtonTapped()
+        }
+    }
+
+    @objc func drawerDidChangeState(state: DrawerState, controller: UIViewController) {
```

Comment:
This is intended for the client to show how does the delegates work. It would encourage them to use this rather than manipulating state. 

Is there any thing wrong with keeping it?

Reply:
I don't think it's necessarily wrong to keep it, but this is an optional method (which wouldn't help us catch func signature changes, right?).
We usually avoid doing just print() since it's not necessarily visible in the demo app.
You can keep it if you're very passionate about it.

Reply:
I have removed it for now, since empty function might look like a bug. We can add this as an example later while adding a mock teams demo controller